src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -0,0 +1,440 @@
+      return DataValidationUtil.validateAndParseDate(type.getName(), value, insertRowsCurrIndex);
+    }
+    throw new SFException(
+        ErrorCode.UNKNOWN_DATA_TYPE, logicalTypeAnnotation, type.getPrimitiveTypeName());
+  }
+
+  /**
+   * Parses an int64 value based on Parquet logical type.
+   *
+   * @param value column value provided by user in a row
+   * @param type Parquet column type
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return parsed int64 value
+   */
+  private static long getInt64Value(
+      Object value, PrimitiveType type, ZoneId defaultTimezone, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return DataValidationUtil.validateAndParseIcebergNumber(
+          type.getName(), value, insertRowsCurrIndex, Long.class);
+    }
+    if (logicalTypeAnnotation instanceof DecimalLogicalTypeAnnotation) {
+      return getDecimalValue(value, type, insertRowsCurrIndex).unscaledValue().longValue();
+    }
+    if (logicalTypeAnnotation instanceof TimeLogicalTypeAnnotation) {
+      return DataValidationUtil.validateAndParseTime(
+              type.getName(),
+              value,
+              timeUnitToScale(((TimeLogicalTypeAnnotation) logicalTypeAnnotation).getUnit()),
+              insertRowsCurrIndex)
+          .longValue();
+    }
+    if (logicalTypeAnnotation instanceof TimestampLogicalTypeAnnotation) {
+      boolean includeTimeZone =
+          ((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).isAdjustedToUTC();
+      return DataValidationUtil.validateAndParseTimestamp(
+              type.getName(),
+              value,
+              timeUnitToScale(((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).getUnit()),
+              defaultTimezone,
+              !includeTimeZone,
+              insertRowsCurrIndex)
+          .toBinary(false)
+          .longValue();
+    }
+    throw new SFException(
+        ErrorCode.UNKNOWN_DATA_TYPE,
+        logicalTypeAnnotation,
+        type.asPrimitiveType().getPrimitiveTypeName());
+  }
+
+  /**
+   * Converts an Iceberg binary or string column to its byte array representation.
+   *
+   * @param value value to parse
+   * @param type Parquet column type
+   * @param stats column stats to update
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return string representation
+   */
+  private static byte[] getBinaryValue(
+      Object value, PrimitiveType type, RowBufferStats stats, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      byte[] bytes =
+          DataValidationUtil.validateAndParseBinary(
+              type.getName(), value, Optional.empty(), insertRowsCurrIndex);
```

Comment:
why not validate against maxLength here?

Reply:
Added max length validation, we don't need the max length in column meta data as the length of binary and string are constants.